lib/benchee/memory_measure.ex

```diff
@@ -0,0 +1,108 @@
+defmodule Benchee.MemoryMeasure do
+  @moduledoc """
+  This exposes two functions, apply/1 and apply/3. Both execute a given function
+  and report on the memory used by monitoring the garbage collection process for
+  a single process.
+  """
+  import Kernel, except: [apply: 3, apply: 2]
+
+  @spec apply(fun) :: no_return() | tuple()
+  def apply(f) do
+    apply(:erlang, :apply, [f, []])
+  end
+
+  @spec apply(atom, atom, list) :: no_return() | tuple()
+  def apply(m, f, a) do
+    ref = make_ref()
+    Process.flag(:trap_exit, true)
+    start_runner(m, f, a, ref)
+
+    receive do
+      {^ref, final} -> final
+      :shutdown -> nil
+    end
+  end
+
+  defp start_runner(m, f, a, ref) do
+    parent = self()
+    word_size = :erlang.system_info(:wordsize)
+
+    spawn_link(fn ->
+      tracer = start_tracer(self())
+
+      try do
+        {:garbage_collection_info, info_before} = Process.info(self(), :garbage_collection_info)
+        result = Kernel.apply(m, f, a)
+        {:garbage_collection_info, info_after} = Process.info(self(), :garbage_collection_info)
+        mem_collected = get_collected_memory(tracer)
+
+        final =
+          {result, (info_after[:heap_size] - info_before[:heap_size] + mem_collected) * word_size}
```

Comment:
Would it make sense to switch the order in the tuple to be consistent with `:timer.tc`?

Reply:
Yeah, that couldn't hurt.